src/router/routes/genericResourceRoute.ts

```diff
@@ -135,23 +155,36 @@ export default class GenericResourceRoute {
                     const resourceType = req.baseUrl.substr(1);
                     const searchParamQuery = req.query;
 
-                    const allowedResourceTypes = await this.authService.getAllowedResourceTypesForOperation({
-                        operation: 'search-type',
-                        userIdentity: res.locals.userIdentity,
-                    });
+                    const updatedReadResponse = await handleSearch(res, resourceType, searchParamQuery);
+                    res.send(updatedReadResponse);
+                }),
+            );
+            this.router.post(
+                '/_search',
+                RouteHelper.wrapAsync(async (req: express.Request, res: express.Response) => {
+                    // Get the ResourceType looks like '/Patient'
+                    const resourceType = req.baseUrl.substr(1);
+                    const searchParamQuery = req.query;
+                    const { body } = req;
 
-                    const response = await this.handler.typeSearch(
-                        resourceType,
-                        searchParamQuery,
-                        allowedResourceTypes,
-                        res.locals.userIdentity,
-                    );
-                    const updatedReadResponse = await this.authService.authorizeAndFilterReadResponse({
-                        operation: 'search-type',
-                        userIdentity: res.locals.userIdentity,
-                        readResponse: response,
+                    const conflictSearchParams: string[] = [];
+                    forOwn(searchParamQuery, (value, key) => {
+                        if (has(body, key) && body[key] !== value) {
```

Comment:
`value` may be an array. express adds duplicate query params into an array

`url/?a=1&?a=2&b=3` results in 
```
req = {
  query: {
    a: [1, 2],
    b: 3
  }
}
```